### ISSUE TITLE
Just 2 questions

### ISSUE THREAD

**Issue by JawadTawhidi** [hamidriasat/UNet-3-Plus#6]
Hi, 

Thanks for making the code of UNet-3-Plus simpler, however I have 2 questions and I would be very grateful to hear from you:

1. In the paper the author says they used ReseNet-101 as backbone encoder, but in your code I could not find where exactly ReseNet-101is used? is it used at all? if not how did you extract the features? using custom convolutional layers? 
2. My second question is about the decoder's logic, did you use custom functions for Max pooling, Relu, batch normalization... etc?

**Comment by hamidriasat** [comment]
Hi @JawadTawhidi 
Thanks for liking my work.

Answers to you questions

1. No, I haven't used ResNet-101 as a backbone. Yes, I used custom convolutional layers in the encoder. [Here](https://github.com/NVIDIA/DeepLearningExamples/tree/master/TensorFlow2/Segmentation/Contrib/UNet3P) you can find using custom layers I was able to reproduce paper results.
2. Yes I have used above mentioned functions in the decoder.

**Comment by JawadTawhidi** [comment]
Ok, good. 

Thank you for responding. I want to use ResNet-101 as backbone for feature extraction part(As the author said in the original paper) but I did not find any similar code. Any clue? Any similar code?

**Comment by hamidriasat** [comment]
I don't have any similar code but to give you a head start, you can use below code to extract features from ResNet-101. Replace them in the encoder section of UNet3+ to extract features.

```
# !pip install keras_cv  # Install KerasCV
import keras_cv
resnet = keras_cv.models.ResNet101Backbone(
        include_rescaling=False,
    )
# ResNet101 blocks output layer names
# {'P2': 'v2_stack_0_block3_out', 'P3': 'v2_stack_1_block4_out', 'P4': 'v2_stack_2_block23_out', 'P5': 'v2_stack_3_block3_out'}
resnet.get_layer("v2_stack_0_block3_out").output
```

**Comment by JawadTawhidi** [comment]
Thank you, but as you know the resnet 101 has 4 layers, I am confused which layers I have to consider as e1, e2, e3...? specially in resnet 101 there is no need to apply max pooling and batch normalization after passing input to each layer. Any help?

**Comment by hamidriasat** [comment]
Above ResNet-101 block output layers names are mentioned, you can use that to extract the output of each block and use them in the encoder.
I am not sure but maybe you have to adjust the network accordingly, right now I do not have time to experiment exactly with what changes will be needed to accommodate ResNet-101.
